### DISCUSSION TITLE
Periodic current spike using tickless Zephyr

### DISCUSSION THREAD

**Discussion by redaarifi** [zephyrproject-rtos/zephyr#53492] [General]
**Describe the bug**
Hello everyone,

I don't believe this to be a bug but rather id like to get some info on what is causing the eddystone zephyr sample to have periodic current spike that reach 20µA. 

 - I'm using an NRF52832 µC
 - Zephyr SDK 0.15.1
 - PPK2 for current measurement.

my proj.conf file config is : 

CONFIG_BT=y
CONFIG_BT_DEBUG_LOG=n
CONFIG_BT_PERIPHERAL=y
CONFIG_BT_DEVICE_NAME="Zephyr Eddystone"

CONFIG_PM=y
CONFIG_PM_DEVICE=y
CONFIG_BOARD_ENABLE_DCDC=y

CONFIG_DEBUG=n
CONFIG_ASSERT=n
CONFIG_DEBUG_OPTIMIZATIONS=n

CONFIG_LOG=n
CONFIG_CONSOLE=n
CONFIG_SERIAL=n
CONFIG_PRINTK=n

CONFIG_LOG_PRINTK=n
CONFIG_EARLY_CONSOLE=n
CONFIG_UART_CONSOLE=n
CONFIG_RTT_CONSOLE=n
CONFIG_USE_SEGGER_RTT=n

CONFIG_GPIO=n

![image](https://user-images.githubusercontent.com/56077385/210580712-86895c10-b1e7-419a-9fe5-bd22f62b278f.png)

**Comment by carlescufi** [comment]
Isn't this the advertising packet going out? CC @cvinayak

**Reply by redaarifi** [reply]
Thank you for the reply. I don't believe it's the case, the advertising packet current consumption is about ~10mA to 15mA. (Refer to the screenshot include below)

The 20µA spikes happens between advertising periods.

Avertising power consumption : 

![image](https://user-images.githubusercontent.com/56077385/210587654-985e4263-3255-44af-afee-88bd075bba15.png)

Current spikes between advertising : 

![image](https://user-images.githubusercontent.com/56077385/210588320-5d7675f0-d3be-4f75-94a9-5200f1bf4ee8.png)

**Comment by cvinayak** [comment]
You are using `CONFIG_BOARD_ENABLE_DCDC=y`, hope you have the required external LC filter connected?

I am guessing the spike are periodic current consumption for the operation of DC/DC regulator.
